src/iris/sender/rpc.py

```diff
@@ -89,16 +89,30 @@ def handle_api_notification_request(socket, address, req):
         logger.warn('Dropping OOB message with invalid target "%s" from app %s',
                     target, notification['application'])
         return
-
-    try:
-        expanded_targets = cache.targets_for_role(role, target)
-    except IrisRoleLookupException:
-        expanded_targets = None
-    if not expanded_targets:
-        reject_api_request(socket, address, 'INVALID role:target')
-        logger.warn('Dropping OOB message with invalid role:target "%s:%s" from app %s',
-                    role, target, notification['application'])
-        return
+    expanded_targets = None
+    # if role is literal_target skip unrolling
+    if role == 'literal_target':
```

Comment:
Do these checks in `Notifications.on_post()` to avoid needless rpc on bad requests.